### DISCUSSION TITLE
Specifying Two `inputDir` and one `outputDir`

### DISCUSSION THREAD

**Discussion by ProfessorManhattan** [hairyhenderson/gomplate#1689] [Q&A]
Hey, I have a have bunch of files required for my repos. All of them inherit files from the `common/` directory and then they also inherit files from a folder named after a variable which is available in the `context`.

How can I configure my `.gomplate.yml` so that the output directory gets all the files from the `common/` directory and then also gets files from an inputDir named after a variable available in the context?

**Comment by hairyhenderson** [comment]
Hi @ProfessorManhattan, I'm not sure I understand the use-case, but from the title it seems you're trying to specify multiple inputDirs. This isn't supported. You'll need to either regroup the input templates into a single directory, or invoke `gomplate` multiple times.

Note that if your directory structure in the input dir doesn't match the layout you want in the output dir, you can use [`outputMap`](https://docs.gomplate.ca/config/#outputmap)

**Comment by saikumar843** [comment]
Hi @hairyhenderson,

I have a very basic requirement to process multiple files from input-dir and create output-map. Following the same structure of the files in output directory and remove the .tmpl suffix from the files in input-dir 

Template 
    Docker
    Scriptsandfiles
    Terraform
        *.tf files
     Pipelines.yaml.tmpl
     Changelog.md.tmpl
     Readme.md.tmpl

When I pass like gomplate --inputdir template output-map=out/{{ .in strings.ReplaceAll ".yaml.tmpl" ".yaml"}} this works only for pipelines.yaml where in .md.tmpl file also needed to be handled in the same gomplate run

Kindly provide some idea

**Reply by hairyhenderson** [reply]
@saikumar843 this looks like a different issue to what was originally reported, so probably belongs in a new discussion!

However, I think you can achieve what you're looking for by using multiple replacements:

```handlebars
--output-map='out/{{ .in | strings.ReplaceAll ".yaml.tmpl" ".yaml" | strings.ReplaceAll ".md.tmpl" ".md" }}'
```

Alternately, you could simplify take advantage of the fact that the templates are all named with `.tmpl` at the end, and use a single replacement to just remove it:

```handlebars
--output-map='out/{{ .in | strings.ReplaceAll ".tmpl" "" }}'
```

**Reply by saikumar843** [reply]
@hairyhenderson Thanks. I should've posted this in a separate discussion.
